lib/colorls/core.rb

```diff
@@ -101,10 +101,12 @@ def filter_contents(path)
       end
     end
 
-    def sort_contents
+    def sort_contents(path)
       case @sort
       when :time
-        @contents.sort_by! { |a| -File.mtime(a).to_f }
+        @contents.sort_by! { |a| -File.mtime(File.join(path,a)).to_f }
```

Comment:
good catch @neshmi ! I have incorporated the `File.join` into my initial commit so that is more legible...